src/bidmanager.js

```diff
@@ -203,34 +184,35 @@ function getKeyValueTargetingPairs(bidderCode, custBidObj) {
 					}
 				}]
 			};
-
-			//set default settings 
-
 		}
+
 		custBidObj.usesGenericKeys = true;
 		setKeys(keyValues, bidder_settings[CONSTANTS.JSON_MAPPING.BD_SETTING_STANDARD], custBidObj);
 	}
 
 	return keyValues;
-
 }
 
 function setKeys(keyValues, bidderSettings, custBidObj) {
 	var targeting = bidderSettings[CONSTANTS.JSON_MAPPING.ADSERVER_TARGETING];
-	custBidObj['size'] = custBidObj.getSize();
-	for (var i = 0; i < targeting.length; i++) {
-		var key = targeting[i].key;
-		var value = targeting[i].val;
-		if (typeof value === objectType_function) {
-			try {
-				keyValues[key] = value(custBidObj);
-			} catch (e) {
-				utils.logError('Exception trying to parse value. Check bidderSettings configuration : ' + e.message);
-			}
-		} else {
+	custBidObj.size = custBidObj.getSize();
+
+  utils._each(targeting, function (kvPair) {
+    var key = kvPair.key,
+        value = kvPair.val;
+
+    if (utils.isFn(value)) {
+      try {
+        keyValues[key] = value(custBidObj);
+      } catch (e) {
+        utils.logError("invalid function during k/v targeting", "BIDMANAGER");
```

Comment:
I like sending the "class" with the error message. Maybe we should add a 3rd param in `utils.logError()` for Exception so we don't lose that info?